### PR TITLE
AI/prompts

### PR DESCRIPTION
**Pull request by ayyubibrahimi**
@marvinmarnold thoughts?

- Created a new faiss index specifically for queries related to voting (a button still needs to be added). The minutes are the only input for this model. It doesn't work atm. It might be worth exploring data extraction using a supervised model with [spacy](https://github.com/explosion/spaCy). It would take some time to build out the model, but it would likely be very reliable. 
- In-Depth and General queries are routed to a template based on the prefix of the query. See below. This line of thinking was inspired by your query "what is a proviso". 
- Still need to explore what margin of error is appropriate for the timestamps

```
def get_in_depth_prompt_template(query):
    defining_prefixes = [
        "what is a",
        "define",
        "explain",
        "what do you know about",
        "tell me about",
        "give information on",
        "describe",
    ]
    defintion_template = """
        As an AI assistant, your task is to explain or define the term or concept "{question}" in the context of the provided transcripts from New Orleans City Council meetings in "{docs}".
        
        Your response should start with a concise explanation or definition of the term or concept. Then, highlight instances or discussions from the transcripts that help illustrate or contextualize the term or concept. If relevant, include any voting actions, but only provide information that is supported by the transcripts.
        
        Note: If the transcripts don't fully cover the scope of the question, it's fine to summarize the available details and state 'The provided documents might not cover all aspects of the question but they do provide some key insights on the following points: 
        Note: If the transcripts don't fully cover the scope of the question or the above-mentioned details are not present in the transcripts, state 'There is insufficient information available to answer this question accurately'.
    """

    general_template = """
        As a language model specifically trained on city council meetings, agendas, and minutes, your objective is to generate an in-depth response to "{question}", using data from the specified New Orleans City Council meeting documents in "{docs}". 
                
        Note: If the transcripts don't fully cover the scope of the question, it's fine to summarize the available details and state 'The provided documents do not cover all aspects of the question but they do provide some key insights on the following points: 
        Should the transcripts lack the relevant details entirely, state "The available information is insufficient for a comprehensive response to this question."
    """

    query = query.lower().strip()
    if any(query.startswith(prefix) for prefix in defining_prefixes):
        return defintion_template
    else:
        return general_template
```

### CODE REVIEW

**Comment by marvinmarnold** [comment]
I still find keyword based routing very fragile and confusing for users. I'd rather just add "define term" to the summary/detailed toggle.

File: packages/googlecloud/functions/getanswer/inquirer.py [hunk 1]
```diff
@@ -218,7 +218,7 @@ def route_question(db_general, db_in_depth, db_voting, query, query_type, k=4):
     elif query_type == RESPONSE_TYPE_GENERAL:
         return get_general_summary_response_from_query(db_general, query, k)
     elif query_type == RESPONSE_TYPE_VOTING:
-        return get_voting_summary_response_from_query(db_voting, query, k=4)
+        return get_voting_summary_response_from_query(db_voting, query, k=1)
```
Comment (marvinmarnold) [inline packages/googlecloud/functions/getanswer/inquirer.py:221]: @ayyubibrahimi heads up that some ordinances go through committee. In which case they will have a vote there too as well.

**Comment by marvinmarnold** [comment]
> I still find keyword based routing very fragile and confusing for users. I'd rather just add "define term" to the summary/detailed toggle.

@ayyubibrahimi I wonder how hard it would be to process the question with a prompt that asks the ML to classify the question as one of the above categories. I'd trust that classification more than a keyword approach.

**Comment by ayyubibrahimi** [comment]
Interesting. I like it. Will look into it.

**Pull request closed without merging**